src/applications/mhv-medications/util/helpers/prescriptionMedAndRenewalStatus.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+import StatusDropdown from '../../components/shared/StatusDropdown';
+
+import {
+  pdfStatusDefinitions,
+  pdfDefaultStatusDefinition,
+  pdfDefaultPendingMedDefinition,
+  pdfDefaultPendingRenewalDefinition,
+} from '../constants';
+
+const determineStatus = (
+  switchCase,
+  pendingMed,
+  pendingRenewal,
+  prescription,
+) => {
+  let statusDefinition;
+  let statusText;
+  switch (switchCase) {
+    case 'VaPrescription':
+      if (pendingRenewal) {
+        statusText = pdfDefaultPendingRenewalDefinition.reduce(
+          (fullStatus, item) =>
+            fullStatus + item.value + (item.continued ? ' ' : '\n'),
+          '',
+        );
+        return <p data-testid="pending-renewal-status">{statusText}</p>;
+      }
+      if (pendingMed) {
+        statusText = pdfDefaultPendingMedDefinition.reduce(
+          (fullStatus, item) =>
+            fullStatus + item.value + (item.continued ? ' ' : '\n'),
+          '',
+        );
+        return <p>{statusText}</p>;
+      }
+      return <StatusDropdown status={prescription?.dispStatus} />;
+    case 'print':
+      if (pendingMed) {
+        statusDefinition = pdfDefaultPendingMedDefinition;
+      } else if (pendingRenewal) {
+        statusDefinition = pdfDefaultPendingRenewalDefinition;
+      } else {
+        statusDefinition =
+          pdfStatusDefinitions[prescription.refillStatus] ||
+          pdfDefaultStatusDefinition;
+      }
+      return statusDefinition.reduce(
+        (fullStatus, item) =>
+          fullStatus + item.value + (item.continued ? ' ' : '\n'),
+        '',
+      );
+    default:
+      return null;
+  }
+};
+
+/**
+ * @param {obj} prescription - The prescription object containing pharmacy phone details.
```

Comment:
```suggestion
 * @param {{prescriptionSource: string, dispStatus: string}} prescription The prescription object containing pharmacy phone details.
 * @param {string} switchCase - Flag to indicate how this status is being displayed
```

nit/non-blocking: total matter of opinion, but for small objects like this I think it's a nice dev experience to define the shape used rather than a generic `obj`

We're also missing the second param, `switchCase`.